substrate/frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1504,11 +1522,12 @@ impl<T: Config> Pallet<T> {
 	/// Parts of [`create_snapshot`] that happen outside of this pallet.
 	///
 	/// Extracted for easier weight calculation.
+	///
+	/// Note: this pallet only supports one page of voter and target snapshots.
 	fn create_snapshot_external(
 	) -> Result<(Vec<T::AccountId>, Vec<VoterOf<T>>, u32), ElectionError<T>> {
 		let election_bounds = T::ElectionBounds::get();
-
-		let targets = T::DataProvider::electable_targets(election_bounds.targets)
+		let targets = T::DataProvider::electable_targets(election_bounds.targets, SINGLE_PAGE)
```

Comment:
You are not using the `T::Pages` here, you probably changed your mind along the way :)) 

Remove the `T::Pages` please, and generally write this pallet with a strict assumption that it only provides a single page implementation.